src/openms/source/ANALYSIS/OPENSWATH/OpenSwathScoring.cpp

```diff
@@ -46,13 +49,55 @@
 
 // basic file operations
 
+namespace OpenMS
+{
+  void sortSpectrumByMZ(OpenSwath::SpectrumPtr spec)
+  {
+    //sort index list
+    std::vector<std::pair<double, Size> > sorted_indices;
+    sorted_indices.reserve(spec->getMZArray()->data.size());
+    auto mz_it = spec->getMZArray()->data.begin();
+    for (Size i = 0; i < spec->getMZArray()->data.size(); ++i)
+    {
+      sorted_indices.push_back(std::make_pair(*mz_it, i));
+      ++mz_it;
+    }
+    std::stable_sort(sorted_indices.begin(), sorted_indices.end(), PairComparatorFirstElement<std::pair<double, Size> >());
+
+    // extract list of indices
+    std::vector<Size> select_indices;
+    select_indices.reserve(sorted_indices.size());
+    for (Size i = 0; i < sorted_indices.size(); ++i)
```

Comment:
prefer range based, since you do not need the index?

Reply:
Don’t know if relevant but nice to know https://stackoverflow.com/questions/1577475/c-sorting-and-keeping-track-of-indexes